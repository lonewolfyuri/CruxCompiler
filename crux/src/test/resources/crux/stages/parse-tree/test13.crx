// test while

func main() : void {
  while (true) {
    ::nothing();
  }

  while 4-5 {
    ::nothing();
  }

}
