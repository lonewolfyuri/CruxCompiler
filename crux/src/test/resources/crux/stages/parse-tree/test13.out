program
  declarationList
    declaration
      functionDefinition
        parameterList
        type
        statementBlock
          statementList
            statement
              whileStatement
                expression0
                  expression1
                    expression2
                      expression3
                        expression0
                          expression1
                            expression2
                              expression3
                                literal
                statementBlock
                  statementList
                    statement
                      callStatement
                        callExpression
                          expressionList
            statement
              whileStatement
                expression0
                  expression1
                    expression2
                      expression3
                        literal
                    op1
                    expression2
                      expression3
                        literal
                statementBlock
                  statementList
                    statement
                      callStatement
                        callExpression
                          expressionList
